src/app/query/stacks/fees/fees.hooks.ts

```diff
@@ -47,63 +55,65 @@ function useFeeEstimationsMinValues() {
+      estimates: getDefaultSimulatedFeeEstimations(txByteLength),
+      calculation: FeeCalculationTypes.DefaultSimulated,
+    };
+  }
 
-export function useStacksFeeEstimations(txByteLength: number | null, txPayload: string) {
-  const feeEstimationsMaxValues = useFeeEstimationsMaxValues();
-  const feeEstimationsMinValues = useFeeEstimationsMinValues();
-  const result = useTransactionFeeEstimation(txByteLength, txPayload);
-  const { data: txFeeEstimation, isError, isLoading } = result;
-
-  return useMemo<FeeEstimations>(() => {
-    const feeEstimations = txFeeEstimation?.estimations;
+  const stacksFeeEstimates: StacksFeeEstimate[] = feeEstimation.estimations.map(estimate => {
+    return {
+      fee: createMoney(estimate.fee, 'STX'),
+      feeRate: estimate.fee_rate,
+    };
+  });
 
-    if (!isLoading && isError) {
-      return {
-        blockchain: 'stacks',
-        estimates: defaultFeeEstimations,
-        calculation: FeeCalculationTypes.Default,
-      };
-    }
-    if (txByteLength && feeEstimationQueryFailedSilently(result)) {
-      return {
-        blockchain: 'stacks',
-        estimates: getDefaultSimulatedFeeEstimations(txByteLength),
-        calculation: FeeCalculationTypes.DefaultSimulated,
-      };
-    }
-    if (feeEstimations?.length) {
-      const feeEstimationsWithCappedValues = getFeeEstimationsWithCappedValues(
-        feeEstimations,
-        feeEstimationsMaxValues,
-        feeEstimationsMinValues
-      );
-      return {
-        blockchain: 'stacks',
-        estimates: feeEstimationsWithCappedValues,
-        calculation: FeeCalculationTypes.FeesCapped,
-      };
-    }
+  if (feeEstimation.estimations && feeEstimation.estimations.length) {
+    const feeEstimationsWithCappedValues = getFeeEstimationsWithCappedValues(
+      stacksFeeEstimates,
+      maxValues,
+      minValues
+    );
     return {
       blockchain: 'stacks',
-      estimates: feeEstimations ?? [],
-      calculation: FeeCalculationTypes.Api,
+      estimates: feeEstimationsWithCappedValues,
+      calculation: FeeCalculationTypes.FeesCapped,
     };
-  }, [
-    feeEstimationsMaxValues,
-    feeEstimationsMinValues,
-    isError,
-    isLoading,
-    result,
-    txByteLength,
-    txFeeEstimation?.estimations,
-  ]);
+  }
+  return {
+    blockchain: 'stacks',
+    estimates: stacksFeeEstimates ?? [],
+    calculation: FeeCalculationTypes.Api,
+  };
+}
+
+export function useStacksFees(txByteLength: number | null, transactionPayload: string) {
+  const feeEstimationsMaxValues = useFeeEstimationsMaxValues();
+  const feeEstimationsMinValues = useFeeEstimationsMinValues();
+
+  return useGetStacksTransactionFeeEstimationQuery(txByteLength, transactionPayload, {
+    onError: () => defaultStacksFees,
+    select: resp =>
+      parseStacksTxFeeEstimationResponse({
+        feeEstimation: resp,
+        maxValues: feeEstimationsMaxValues,
+        minValues: feeEstimationsMinValues,
+        txByteLength,
+      }),
+  });
```

Comment:
Naming wise I don't think it's clear the difference in role between `useStxFees` and `useStacksFees`

Reply:
Personally, I just think people need to get used to naming in the code now where STX means the native asset and `Stacks` means the blockchain? It is clear to me.

Reply:
I guess I just find it not descriptive enough. In isolation, what is "stacks fees"? Maybe something like `useCalculateStacksTxFee` would be clearer.